### PR TITLE
Working around file timeout issues

### LINKED ISSUE #8486: File not found during parallel runs
When running with parallel options with the TestHarness, we are running into an event where not all of the output are available for reading:

```pre
misc/check_error.bad_kernel_test.......................................................................... OK
misc/check_error.bad_kernel_var_test...................................................................... OK
Traceback (most recent call last):
  File "./run_tests", line 24, in <module>
    TestHarness.buildAndRun(sys.argv, app_name, MOOSE_DIR)
  File "/home/moosetest/civet/client_root_0/moose/python/TestHarness/TestHarness.py", line 36, in buildAndRun
    harness.findAndRunTests()
  File "/home/moosetest/civet/client_root_0/moose/python/TestHarness/TestHarness.py", line 233, in findAndRunTests
    self.runner.run(tester, command)
  File "/home/moosetest/civet/client_root_0/moose/python/TestHarness/RunParallel.py", line 79, in run
    self.startReadyJobs(slot_check)
  File "/home/moosetest/civet/client_root_0/moose/python/TestHarness/RunParallel.py", line 150, in startReadyJobs
    self.run(tester, command, recurse=False, slot_check=slot_check)
  File "/home/moosetest/civet/client_root_0/moose/python/TestHarness/RunParallel.py", line 103, in run
    self.spinwait()
  File "/home/moosetest/civet/client_root_0/moose/python/TestHarness/RunParallel.py", line 221, in spinwait
    self.returnToTestHarness(job_index)
  File "/home/moosetest/civet/client_root_0/moose/python/TestHarness/RunParallel.py", line 174, in returnToTestHarness
    output += self.getOutputFromFiles(tester)
  File "/home/moosetest/civet/client_root_0/moose/python/TestHarness/RunParallel.py", line 159, in getOutputFromFiles
    with open(file_path, 'r') as f:
IOError: [Errno 2] No such file or directory: '/home/moosetest/civet/client_root_0/moose/test/tests/misc/check_error/Badd_material_test.processor.1'
ERROR: exiting with code 1
```
Upon logging into the machine in question, the file is indeed available. This seems to only be replicable on my CentOS 7 VM. Each time being a different test.

### PR DESCRIPTION
**Pull request by permcody**
closes #8486

This will hopefully fix the issue where we get errors trying to read from files that have been redirected. No rush on merging this. I haven't even come up with a way of testing this robustly. 

Note: I did return False in the new checkOutputReady() method to see if it would report the FILE TIMEOUT error, which it did. I may just need to build in a testing branch in that method under special conditions. (tag @rwcarlsen, @brianmoose for ideas).

### CODE REVIEW

File: python/TestHarness/RunParallel.py [hunk 1]
```diff
@@ -234,19 +252,37 @@ def spinwait(self, time_to_wait=0.05):
         for tuple in self.jobs:
             if tuple != None:
                 (p, command, tester, start_time, f, slots) = tuple
-                if p.poll() != None or now > (start_time + float(tester.specs['max_time'])):
-                    # finish up as many jobs as possible, don't sleep until
-                    # we've cleared all of the finished jobs
-                    self.returnToTestHarness(job_index)
+                # Look for completed tests
+                if p.poll() != None:
+                    if not self.checkOutputReady(tester):
```
Comment (permcody) [inline python/TestHarness/RunParallel.py]: This is the key with this PR. We test to make sure that the output is available and if not we leave the job in the jobs queue for awhile longer (self.default_reporting_time or 10 seconds).

**Comment by permcody** [comment]
Tag @milljm for potential testing before merge

**Comment by permcody** [comment]
@milljm - I just doubled the timeout and also added the `didFail()` method to the Tester class along with a lot of documentation about why both `didPass` and `didFail`both exist. If you are satisfied with this PR, we can merge it in. Maybe we can test this a bit more to see if the longer timeout helps? Oh also fixed the unlikely case of having a reported test not receiving the full second timeout.

**Comment by permcody** [comment]
We've been testing this and the PR appears to be working properly since we've actually been able to catch the "not ready" problem on Jason's VM. I do not have a test for this new case but it's rare and a big improvement over the TestHarness crashing (which it does now).

**Comment by permcody** [comment]
OK - New theory...

On @milljm's test system we are seeing that in some cases, one of the files from a parallel run is _never_ written. These are happening on the check_error tests which have errors that get triggered very early on in the execution. I'm thinking that on an overloaded machine, it might be possible for one or more ranks to never get an opportunity to execute before another rank has already crashed. If this theory is correct, that file may never be created!

This is a difficult situation to test, but I'm thinking we relax the TestHarness a bit and allow it to read only the files available after the timeout and allow the Tester to decide if the results are valid or not.

**Comment by permcody** [comment]
@milljm - has confirmed that this solves the failure problem on his VM.

**Review by aeslaughter — APPROVED** [review]

**Pull request merged**